src/main/java/mate/academy/dao/BookDaoimpl.java

```diff
@@ -0,0 +1,121 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoimpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO book(title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement ps = connection.prepareStatement(sql,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            ps.setString(1, book.getTitle());
+            ps.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = ps.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at least one row, but inserted "
+                        + affectedRows + " rows");
+            }
+            ResultSet rs = ps.getGeneratedKeys();
+            if (rs.next()) {
+                Long id = rs.getLong(1);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book: " + book, e);
```

Comment:
Use a more informative message for the exception. For example: "Can't add new book: " + book + " to the database."